Data/Vector/Generic/Mutable.hs

```diff
@@ -620,6 +627,26 @@ replicateM :: (PrimMonad m, MVector v a) => Int -> m a -> m (v (PrimState m) a)
 {-# INLINE replicateM #-}
 replicateM n m = munstream (MBundle.replicateM n m)
 
+-- | /O(n)/ Create a mutable vector of the given length (0 if the length is negative)
+-- and fill it with the results of applying the function to each index.
+generate :: (PrimMonad m, MVector v a) => Int -> (Int -> a) -> m (v (PrimState m) a)
+{-# INLINE generate #-}
+generate n f = stToPrim $ generateM n (return . f)
```

Comment:
It allows GHC to work in the ST monad instead of whatever tower of monad transformers is thrown at it. GHC may or may not be able to optimize those binds. Hopefully this will reduce amount of work optimizer has to do and may even result in better code. Same for other function that take pure function as parameters